doc/hardware/peripherals/i3c.rst

```diff
@@ -0,0 +1,338 @@
+
+      * If there is a device waiting for address, it will send
+        its Provisioned ID, BCR, and DCR back. Match the received
+        Provisioned ID to the list of registered I3C devices.
+
+        * If there is a match, assign an address (either from
+          the stated static address if ``SETDASA`` has not been
+          done, or use a free address).
+
+          * Also, set the BCR and DCR fields in the device descriptor
+            struct.
+
+        * If there is no match, depending on policy, it can be
+          assigned a free address, or the device driver can stop
+          the assignment process and errors out.
+
+          * Note that the I3C API requires device descriptor to
+            function. A device without a device descriptor cannot be
+            accessed through the API.
+
+      * This step can be skipped if there is no connected devices
+        requiring DAA.
+
+   #. These are optional but highly recommended:
+
+      * Do ``GETMRL`` and ``GETMWL`` to get maximum read/write
+        length.
+
+      * Do ``GETMXDS`` to get maximum read/write speed and maximum
+        read turnaround time.
+
+In-Band Interrupt (IBI)
+-----------------------
+
+If a target device can generate In-Band Interrupt (IBI),
+the controller needs to be made aware of it.
+
+* :c:func:`i3c_ibi_slot_request` to request/allocate an IBI slot
+  in the controller.
+
+  * This sets up the IBI slot so the controller can recognize
+    incoming IBI from target devices.
+
+  * This only programs the IBI slot, and the IBI is not active.
+    Call :c:func:`i3c_ibi_enable` to activate the slot.
+
+  * Note that there are usually limited IBI slots on
+    the controller so this operation may fail.
+
+* :c:func:`i3c_ibi_slot_free` to free a previosly allocated IBI
+  slot. Use this when IBI is no longer needed to recognized by
+  the controller.
+
+* :c:func:`i3c_ibi_enable` enables the controller to raise event(s)
+  when there is an incoming IBI.
+
+* :c:func:`i3c_ibi_disable` tells controller not to raise any
+  event(s) where there is incoming IBIs.
+
+Device Tree
+-----------
+
+Here is an example for defining a I3C controller in device tree:
+
+.. code-block:: devicetree
+
+   i3c0: i3c@10000 {
+           compatible = "vendor,i3c";
+
+           #address-cells = < 0x3 >;
+           #size-cells = < 0x0 >;
+
+           reg = < 0x10000 0x1000 >;
+           interrupts = < 0x1F 0x0 >;
+
+           pinctrl-0 = < &pinmux_i3c >;
+           pinctrl-names = "default";
+
+           i2c-scl-hz = < 400000 >;
+
+           i3c-scl-hz = < 12000000 >;
+
+           status = "okay";
+
+           i3c_dev0: i3c_dev0@420000ABCD12345678 {
+                   compatible = "vendor,i3c_dev";
+
+                   reg = < 0x42 0xABCD 0x12345678 >;
+
+                   status = "okay";
+           };
+
+           i2c_dev0: i2c_dev0@380000000000000050 {
+                   compatible = "vendor_i2c_dev";
```

Comment:
I2C devices will not work when they are defined as a direct child of the i3c bus. I also don't think it's a good idea for any future drivers to directly implement the i3c_i2c api calls... 
The I2C virtual controller is the way to go for i2c device drivers on an i3c bus for now...